### PR TITLE
MRG: Add data fetcher for brainstorm data

### PR DESCRIPTION
**Pull request by mainakjas**
cc @agramfort 

they have multiple datasets (Dataset auditory, Dataset resting, Dataset Median nerve CTF). Should we download all of them at once and keep them in subfolders and then make `data_path` point to the top level folder?

TODOs

- [x] add license prompt
- [x] DESCR attribution (like in sklearn)
- [x] add example
- [x] update test

### CODE REVIEW

File: mne/datasets/utils.py [hunk 1]
```diff
@@ -143,6 +145,7 @@ def _data_path(path=None, force_update=False, update_path=True, download=True,
         sample='MNE-sample-data-processed.tar.gz',
         spm='MNE-spm-face.tar.bz2',
         somato='MNE-somato-data.tar.gz',
+        brainstorm='sample_ctf',
```
Comment (mainakjas) [inline mne/datasets/utils.py]: this will be changed. I have this just because it's the smallest file to test and I'm on a slow connection

**Comment by Eric89GXL** [comment]
Look at what we do with the `megsim` dataset. All data are stored under the same folder, but we only download specific files on demand.

**Comment by Eric89GXL** [comment]
BTW for some reason that commit does not point to your github account appropriately, did you develop it on a new machine/setup with the email not set properly?

**Comment by mainakjas** [comment]
Ah, I think it's not added as a verified email. I modified the email address of the commit for now

**Comment by agramfort** [comment]
cool !

regarding the example first make a basic one that just computes Evoked data.

For source analysis we'll need to ship our own freesurfer recon etc...

**Comment by mainakjas** [comment]
I'm not quite sure why travis failed. Any clue @Eric89GXL @agramfort ?

@agramfort the datasets here are not objects like in sklearn, so I'm not sure what to do about the DESCR attribute.

Other than that, I think it's good to go. Please test it at your end before merging.

**Comment by Eric89GXL** [comment]
your fix looks like it should work

File: doc/whats_new.rst [hunk 2]
```diff
@@ -63,6 +60,13 @@ BUG
 
     - Fix dropping of events after downsampling stim channels by `Marijn van Vliet`
 
+API
+~~~
+
+    - Deprecated :class:`mne.decoding.transformer.ConcatenateChannels` and replaced by :class:`mne.decoding.transformer.EpochsVectorizer` by `Romain Trachel`_ 
+
+    - Deprecated `lws` and renamed `ledoit_wolf` for the ``reg`` argument in :class:`mne.decoding.csp.CSP` and :class:`mne.preprocessing.Xdawn` by `Romain Trachel`_ 
+
```
Comment (agramfort) [inline doc/whats_new.rst]: Xdawn is not been released so don't need to document deprecation.

**Comment by agramfort** [comment]
I remember now that for SPM data that is also CTF I put on mne-scripts the scripts we used to generate the dataset we have now on amazon.

https://github.com/mne-tools/mne-scripts/tree/master/spm_face

maybe we could do the same as it would allow to allow ship the Freesurfer recon

**Comment by agramfort** [comment]
regarding DESCR we could have a description function that we could call with

brainstorm.description()

and that prints the details about the data. Now this suggests that we need a path or module for each dataset (auditory, etc...)

**Comment by mainakjas** [comment]
@agramfort description method added + recon files in updated dataset now. Give it a try!

**Comment by agramfort** [comment]
it's downloading. Is it expected not to see a progress bar?

**Comment by mainakjas** [comment]
no, I do see a progress bar -- on both my workstation and my laptop. Weird that you don't see a progress bar

**Comment by agramfort** [comment]
ok now I have it. It's because I was using mne.set_log_level('WARNING')

File: mne/datasets/brainstorm/bst_auditory.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+def description(verbose=None):
+    """Get description of brainstorm (bst_auditory) dataset
+
+    Parameters
+    ----------
+    verbose : bool, str, int, or None
+        If not None, override default verbose level (see mne.verbose).
+    """
+    for desc in _description.splitlines():
+        logger.info(desc)
```
Comment (agramfort) [inline mne/datasets/brainstorm/bst_auditory.py]: I would just use print here. It's weird not to see anything if set_log_level is set to WARNING.

**Comment by agramfort** [comment]
we should have an example for all the datasets we can ship.

**Comment by agramfort** [comment]
how does the evoked compare to the brainstorm one?

**Comment by mainakjas** [comment]
now we have three cousins to take care of :) appveyor isn't happy for some weird reason. Anyone has any clue? I don't have a windows machine at my disposal ...

**Comment by mainakjas** [comment]
@agramfort something fishy here:
Here's what BST people get: 
bst_topo
and here's what we get
mne_topo
even if we disregard the fact that the bst topos don't have uniform scaling, it looks a bit different to my eye. The butterfly plots look like this:
bst butterfly (has stim artifact): 
offsetdone
our butterfly (stim artifact fixed):
mne_butterfly

**Comment by mshamalainen** [comment]
If this is right-hand somatosensory stimulation, ours looks more correct. 

What is the detailed course of the maps from, say, 10 to 35 ms?

- Matti

> On Aug 5, 2015, at 4:53 PM, mainakjas <notifications@github.com> wrote:
> 
> @agramfort <https://github.com/agramfort> something fishy here:
> Here's what BST people get: 
>  <https://cloud.githubusercontent.com/assets/3817535/9097433/384dfcc4-3be1-11e5-8e48-bb97c86e27fc.png>
> and here's what we get
>  <https://cloud.githubusercontent.com/assets/3817535/9097451/5c27d7a0-3be1-11e5-8de9-a23b904b6794.png>
> even if we disregard the fact that the bst topos don't have uniform scaling, it looks a bit different to my eye. The butterfly plots look like this:
> bst butterfly (has stim artifact): 
>  <https://cloud.githubusercontent.com/assets/3817535/9097500/a84559a0-3be1-11e5-9bd9-cd3844473d97.gif>
> our butterfly (stim artifact fixed):
>  <https://cloud.githubusercontent.com/assets/3817535/9097549/e7217780-3be1-11e5-9bc6-1764e29ff49d.png>
> —
> Reply to this email directly or view it on GitHub <https://github.com/mne-tools/mne-python/pull/2358#issuecomment-128144778>.
> 



The information in this e-mail is intended only for the person to whom it is
addressed. If you believe this e-mail was sent to you in error and the e-mail
contains patient information, please contact the Partners Compliance HelpLine at
http://www.partners.org/complianceline . If the e-mail was sent to you in error
but does not contain patient information, please contact the sender and properly
dispose of the e-mail.

**Comment by Eric89GXL** [comment]
It looks to my eye like you can actually get their activation maps if you take a specific subset from our topos (with a translation and *maybe* rotation), I wonder if we're computing our virtual head location or so differently.

**Comment by mainakjas** [comment]
@mshamalainen was spot on! Ours is indeed right-hand somato stimulation whereas the bst example I was looking at was left-hand somato. Here's our topomap for left-hand somato

mne_leftsomato

Now I think it's directly comparable. Sorry for the goof up and good catch indeed.

**Comment by mainakjas** [comment]
... and here' the detailed time course (ours) for left-hand vs right-hand  somato for 10 to 35 ms
left hand:
detailed_leftsomato
right hand:
detailed_right-hand

**Comment by agramfort** [comment]
can you show the butterfly plot with the same time range for both datasets
with bst and MNE?

**Comment by mainakjas** [comment]
Here you go
mne:
mne_average

bst:
bst_average

The only difference is that we fix the stim artifact.

**Comment by mainakjas** [comment]
and the bst topoplot:
bst_topoplot
vs mne topoplot:
mne_topoplot
using the same duration for baseline correction. On second thoughts, they look quite similar to me.

I think we're good.

File: mne/datasets/utils.py [hunk 4]
```diff
@@ -59,6 +59,25 @@
 """
 
 
+_license_text = """
```
Comment (agramfort) [inline mne/datasets/utils.py]: call it _bst_license_text. Why is it in utils and not in a brainstorm file?

File: doc/whats_new.rst [hunk 5]
```diff
@@ -73,6 +67,10 @@ API
 
     - Rename and deprecate ``mne.viz.plot_topo`` for ``mne.viz.plot_evoked_topo`` by `Jaakko Leppakangas`_
 
+    - Deprecated :class:`mne.decoding.transformer.ConcatenateChannels` and replaced by :class:`mne.decoding.transformer.EpochsVectorizer` by `Romain Trachel`_ 
+
+    - Deprecated `lws` and renamed `ledoit_wolf` for the ``reg`` argument in :class:`mne.decoding.csp.CSP` by `Romain Trachel`_
```
Comment (agramfort) [inline doc/whats_new.rst]: these changes are unrelated to this PR and may have been fixed mean while

File: doc/python_reference.rst [hunk 6]
```diff
@@ -237,6 +237,20 @@ Sample datasets
...
+ :no-members:
+ :no-inherited-members:
+
+.. currentmodule:: mne.datasets.brainstorm
+
+.. autosummary::
+   :toctree: generated/
+   :template: function.rst
+
+   data_path
```
Comment (agramfort) [inline doc/python_reference.rst]: don't you have a data_path for all bst datasets?

Comment (mainakjas) [inline doc/python_reference.rst]: woops, I need to update this.

Comment (agramfort) [inline doc/python_reference.rst]: ping me when it's done.

File: mne/datasets/utils.py (hunk 4 above)
Comment (mainakjas) [inline mne/datasets/utils.py]: it's here because the bulk of the download happens here, including asking the user to agree to the license. The brainstorm files don't have much in them ...

**Comment by agramfort** [comment]
Thanks !
